pallets/asset/src/lib.rs

```diff
@@ -1854,17 +1876,26 @@ impl<T: Config> Module<T> {
         Ok(())
     }
 
-    fn base_redeem(origin: T::Origin, ticker: Ticker, value: Balance) -> DispatchResult {
+    fn base_redeem(
+        origin: T::Origin,
+        ticker: Ticker,
+        value: Balance,
+        portfolio_kind: PortfolioKind,
+    ) -> DispatchResult {
         // Ensure origin is agent with custody and permissions for default portfolio.
-        let agent = Self::ensure_agent_with_custody_and_perms(origin, ticker)?;
+        let agent = Self::ensure_agent_with_custody_and_perms(origin.clone(), ticker)?;
 
         Self::ensure_granular(&ticker, value)?;
 
         // Reduce caller's portfolio balance. This makes sure that the caller has enough unlocked tokens.
         // If `advance_update_balances` fails, `reduce_portfolio_balance` shouldn't modify storage.
-        let agent_portfolio = PortfolioId::default_portfolio(agent);
+        let portfolio = PortfolioId {
+            did: agent,
+            kind: portfolio_kind,
+        };
```

Comment:
We should also check if the portfolio exists.  Before we didn't need the check because the user always has a `Default` portfolio.  You can use the `ensure_portfolio_validity` method from the portfolios pallet.

Also the error `PortfolioDoesNotExist` will need to be added to the list of errors for the extrinsic.